Source/WebInspectorUI/UserInterface/Controllers/JavaScriptRuntimeCompletionProvider.js

```diff
@@ -160,9 +172,11 @@ WI.JavaScriptRuntimeCompletionProvider = class JavaScriptRuntimeCompletionProvid
 
         function updateLastPropertyNames(propertyNames)
         {
-            if (this._clearLastPropertiesTimeout)
-                clearTimeout(this._clearLastPropertiesTimeout);
-            this._clearLastPropertiesTimeout = setTimeout(this._clearLastProperties.bind(this), WI.JavaScriptLogViewController.CachedPropertiesDuration);
+            if (this._clearCachedPropertyNamesTimeout)
+                clearTimeout(this._clearCachedPropertyNamesTimeout);
+            // Clears the cache of property names to force picking up any changes occurred while stepping, executing new code in the console prompt, or sitting idle.
+            // 
```

Comment:
Whoops, I think I meant to create a bug and write a FIXME here about the 30-second thing